linkerd/app/outbound/src/endpoint.rs

```diff
@@ -258,8 +257,10 @@ impl<P> CanOverrideAuthority for Endpoint<P> {
 }
 
 impl tap::Inspect for HttpEndpoint {
-    fn src_addr<B>(&self, _: &http::Request<B>) -> Option<SocketAddr> {
-        None
+    fn src_addr<B>(&self, req: &http::Request<B>) -> Option<SocketAddr> {
+        req.extensions()
+            .get::<ClientAddr>()
+            .map(|c| c.as_ref().clone())
```

Comment:
nit: `SocketAddr` is `Copy`, so I think this could just be  
```suggestion
            .map(|c| c.as_ref().copied())
```

Reply:
or, maybe we should just have an `Into<SocketAddr>` impl for `&'_ ClientAddr`?